### PR TITLE
add usefull day access for DateTime

### PR DESCRIPTION
**Pull request by ArthurREGNARD**
- Thanks for your contribution! Please replace this text with a description of what this PR is changing or adding and why, list any relevant issues, and review the contribution guidelines below.

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/sdk/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.

Note that this repository uses Gerrit for code reviews. Your pull request will be automatically converted into a Gerrit CL and a link to the CL written into this PR. The review will happen on Gerrit but you can also push additional commits to this PR to update the code review.
</details>

### CODE REVIEW

File: sdk/lib/core/date_time.dart [hunk 1]
```diff
@@ -938,4 +938,36 @@ extension DateTimeCopyWith on DateTime {
...
+
+  /// Creates a new [DateTime] from this one by only keep year, month and
+  /// add 1 day duration to day.
+  ///
+  /// Example:
+  /// ```dart
+  /// final tomorrow = DateTime.now().toNextDay();
+  /// ```
+  DateTime toNextDay() {
+    return toDay().add(Duration(days: 1));
```
Comment (albertms10) [inline sdk/lib/core/date_time.dart:960]: Don't know whether it’s more performant to increase/decrease the `day` part by 1:

```suggestion
    return DateTime(this.year, this.month, this.day - 1);
```

File: sdk/lib/core/date_time.dart [hunk 2]
```diff
@@ -938,4 +938,36 @@ extension DateTimeCopyWith on DateTime {
...
+  }
+
+  /// Creates a new [DateTime] from this one by only keep year, month and
+  /// substract 1 day duration to day.
+  ///
+  /// Example:
+  /// ```dart
+  /// final yesterday = DateTime.now().toPrevDay();
+  /// ```
+  DateTime toPrevDay() {
```
Comment (alestiago) [inline sdk/lib/core/date_time.dart:970]:
```suggestion
  DateTime toPreviousDay() {
```

Comment (ArthurREGNARD) [reply sdk/lib/core/date_time.dart]: Done

File: sdk/lib/core/date_time.dart (hunk 1 above)
Comment (ArthurREGNARD) [reply sdk/lib/core/date_time.dart]: Done

**Comment by lrhn** [comment]
After more thought here, I think I'll prefer the longer-term strategy of introducing a `Date` extension type which wraps UTC `DateTime`s with zero time-parts. That allows giving it its own API, without needing to fit it into the `DateTime` API.
(I've also already written one, just to be sure it's possible.)

So, not going to add members like this just now.

**Comment by mraleph** [comment]
Closing based on @lrhn answer.

**Pull request closed without merging**